test/test_sparse.py

```diff
@@ -1453,6 +1453,21 @@ def test_shape(num_mats, dim_i, dim_j, dim_k, nnz):
         test_shape(10, 10, 100, 0, 20)
         test_shape(10, 10, 100, 0, 20)
 
+    @onlyCUDA
+    @unittest.skipIf(
+        IS_WINDOWS and TEST_CUDA,
+        "bmm sparse-dense CUDA is not yet supported in Windows, at least up to CUDA 10.1"
+    )
+    def test_bmm_oob(self, device):
```

Comment:
It should be pretty simple to add this as a case for [`test_bmm` above](https://github.com/pytorch/pytorch/blob/ea42027e0ed7530386ae4222dc599fcaf84a8a05/test/test_sparse.py#L1367)

Since you need one sub matrix with no nonzero values, adding a test case with a single nnz and `num_mats > 1` should be able to trigger it?

Reply:
Would also prefer not to have a separate test case. The thing that makes this a bit tricky is that the error is silent if the memory before the index tensor has been allocated since CUDA doesn't do any bounds checking (hence the call to `torch.cuda.empty_cache` at the beginning of the test). So we need to be extra careful in how we set things up in order for the error to trigger reliably (can't use `self._gen_sparse`). For example, just copy-pasting the current test to the end of `test_bmm` results in the exception no longer being raised. This can be fixed by either having the test at the top of `test_bmm` or keeping it at the end but prepending it with `del a, b, ab, ab_mat, ab_mat_check, ab_traspose_check`. However, I feel both solutions could easily break in the future, so I'm not sure what the best option would be?

Reply:
Ah, I didn't realize that it was so sensitive. In that case a stand-alone test like this is warranted.

Reply:
What about a comment explaining the need  for a standalone test? Seems rather non-trivial

Reply:
Thanks for the review @nikitaved! I've added a comment to the test